src/extremeweatherbench/calc.py

```diff
@@ -240,3 +305,38 @@ def nantrapezoid(
         y = np.asarray(y)
         ret = np.add.reduce(d * (y[tuple(slice1)] + y[tuple(slice2)]) / 2.0, axis)
     return ret
+
+
+def compute_specific_humidity_from_relative_humidity(data: xr.Dataset) -> xr.DataArray:
+    """Compute specific humidity from relative humidity and air temperature.
+
+    Args:
+        data: The xarray dataset to compute the specific humidity from containing
+        level (hPa), air_temperature (Kelvin), and relative_humidity. If level is not
+        included in the dataset, assumed to be surface pressure.
+
+    Returns:
+        A DataArray of specific humidity.
+    """
+
+    # Check that the required variables are in the dataset
+    if "air_temperature" not in data.data_vars:
+        raise ValueError("air_temperature must be in the dataset")
+    if "relative_humidity" not in data.data_vars:
+        raise ValueError("relative_humidity must be in the dataset")
```

Comment:
I still think that these should be the actual inputs and you shouldn't pass around arbitrary `xr.Dataset`s where you constantly need to be checking this.